api/client/contextdialer.go

```diff
@@ -78,20 +79,65 @@ func newTunnelDialer(ssh ssh.ClientConfig, keepAlivePeriod, dialTimeout time.Dur
 			return nil, trace.Wrap(err)
 		}
 
-		ssh.Timeout = dialTimeout
-		sconn, err := sshutils.NewClientConnWithDeadline(conn, addr, &ssh)
+		sconn, err := sshConnect(conn, ssh, dialTimeout, addr)
 		if err != nil {
-			return nil, trace.NewAggregate(err, conn.Close())
+			return nil, trace.Wrap(err)
```

Comment:
Does `conn` still need to be closed in the error case?

Reply:
no, the conn.Close call is invoked inside the `sshConnect` func call in case of an error. 
```
	sconn, err := sshutils.NewClientConnWithDeadline(conn, addr, &ssh)
	if err != nil {
		return nil, trace.NewAggregate(err, conn.Close())
	}


	conn, _, err = sshutils.ConnectProxyTransport(sconn.Conn, &sshutils.DialReq{
		Address: constants.RemoteAuthServer,
	}, true)
	if err != nil {
		return nil, trace.NewAggregate(err, sconn.Close())
	}
	```